.github/workflows/build-release.yml

```diff
@@ -0,0 +1,74 @@
+name: build
+on:
+  push:
+    branches: ["main"]
+    tags: ["*"]
+  pull_request:
+
+jobs:
+  tests:
+    name: ${{ matrix.name }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { name: "3.8", python: "3.8", tox: py38 }
+          - { name: "3.12", python: "3.12", tox: py312 }
+    steps:
+      - uses: actions/checkout@v4.0.0
+      - uses: actions/setup-python@v4.7.0
+        with:
+          python-version: ${{ matrix.python }}
+          allow-prereleases: true
+      - run: python -m pip install --upgrade pip
+      - run: python -m pip install tox
+      - run: python -m tox -e ${{ matrix.tox }}
+  build:
+    if: startsWith(github.ref, 'refs/tags')
```

Comment:
@sloria it's better to build the dists unconditionally so that any problems show up early, not last minute, during the release. Also, it's useful to add `twine check --strict`. My action that you use for publishing, runs metadata linting so it can be unpleasant to get the release fail on the very last step. This would help keep the metadata healthy.